var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -763,3 +824,104 @@ def check(self):
         with spack.util.environment.set_env(CP2K_DATA_DIR=data_dir, PWD=self.build_directory):
             with working_dir(self.build_directory):
                 make("test", *self.build_targets)
+
+
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    """Use the new cmake build system to build cp2k. It is the default when
+    building the master branch of cp2k."""
+
+    def cmake_args(self):
+        spec = self.spec
+
+        if "+cuda" in spec and len(spec.variants["cuda_arch"].value) > 1:
+            raise InstallError("CP2K supports only one cuda_arch at a time")
+
+        if "+rocm" in spec and len(spec.variants["amdgpu_target"].value) > 1:
+            raise InstallError("CP2K supports only one amdgpu_arch at a time")
+
+        args = []
+
+        if spec.satisfies("+cuda") or spec.satisfies("+rocm"):
+            if spec.satisfies("cuda_target"):
```

Comment:
```suggestion
            if not spec.satisfies("cuda_arch=none"):
```
?

_But_, it might be clearer to also have separate branches for `spec.satisfies("+cuda")` and `spec.satisfies("+rocm")` (possibly even merging them with the `InstallError` checks above and the `CP2K_USE_ACCEL` branches below?